app/services/encryption/password_verifier.rb

```diff
@@ -14,7 +14,7 @@ def self.parse_from_string(digest_string)
           data[:password_salt],
           data[:password_cost]
         )
-      rescue JSON::ParserError
+      rescue JSON::ParserError, TypeError
```

Comment:
I don't think this is the correct diagnosis of this error. I think we have an incident in production. I see a bunch of legitimate USAJOBS users running into this. Take a look at the Exception Notification logs in our Google group.

Reply:
Yes.  I see it.  I'll remove this one.  I'm surprised it took 4 days to notice.

Reply:
Are you sure raising an encryption error doesn't actually fix the problem?

Reply:
We should probably be guarding against `nil` in `.verify` instead.

E.g.

```ruby
def verify(digest, password)
  return false if password.empty?
  # ...
end
```

Reply:
Sounds reasonable.

Reply:
I think I'll leave the TypeError rescue in as well.